server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -105,13 +121,45 @@ public RemoteFsTranslog(
         BooleanSupplier startedPrimarySupplier,
         RemoteTranslogTransferTracker remoteTranslogTransferTracker,
         RemoteStoreSettings remoteStoreSettings
+    ) throws IOException {
+        this(
+            config,
+            translogUUID,
+            deletionPolicy,
+            globalCheckpointSupplier,
+            primaryTermSupplier,
+            persistedSequenceNumberConsumer,
+            blobStoreRepository,
+            threadPool,
+            startedPrimarySupplier,
+            remoteTranslogTransferTracker,
+            remoteStoreSettings,
+            0
+        );
+    }
+
+    public RemoteFsTranslog(
```

Comment:
Would it be possible to have a new extended class with overridden functionalities for supporting new metadata file format and/or pinned timestamp?

Reply:
We don't need to handle new metadata file format separately. It would work with existing implementation as well. Major changes are made due to pinned timestamp and its handling in GC flow.

IMO, we can still have GC flow with and without pinned timestamp in the same class but if we think this is getting into unmaintainable code, we can think about extending.